### DISCUSSION TITLE
Is it possible to include all local changes in the _changes feed?

### DISCUSSION THREAD

**Discussion by sinm** [apache/couchdb#3327] [Ideas] [answered]
Since v.2 some document updates get omitted from `_changes` feed. While this is ok for a cluster/replication behaviour in general, it would be very nice to see all the changes for every update done **locally**. Maybe in a separate stream. Maybe for revisions introduced by `_update` functions only.

E.g. this feature will allow background post-processing based on specific state of a document by third party services. We can't trigger such processing in a view server or in an `_update` function in context of un-commited document.

**Comment by rnewson** [comment, marked as answer]
It is not possible, because CouchDB does not record them. The `_changes` feed is the same output as the `_all_docs` endpoint except it shows deleted docs too and is in update order. With `feed=continuous` mode you can also listen in on changes as they happen, though there's no guarantee every update to a doc is sent, and you would not see them again on a subsequent call. CouchDB is tracking only the latest revisions of the document, older revisions are available only until next compaction and are considered obsoleted by the later updates.
